backend/services/services.py

```
@@ -60,26 +99,145 @@
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=f"Invalid date/time format: {str(e)}")
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=f"Error processing date/time: {str(e)}")
+
+def try_convert_datetime(date: str, time: str, tz: str, fallback: datetime) -> datetime:
+    
+    if not all([date, time, tz]):
+        return fallback
+    return convert_to_utc(date, time, tz)
 
+def update_fields(fields: Dict[str, Any], model_instance: Any, db: Session) -> bool:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)